Sources/AppCoinsSDK/UI/Purchase/ViewModels/BottomSheetViewModel.swift

```diff
@@ -54,6 +54,16 @@ internal class BottomSheetViewModel: ObservableObject {
     
     @Published var isCreditCardView: Bool = false
     
+    @Published var canChooseMethod: Bool = false
```

Comment:
General comment regarding these variables:
1. They should all be transferred to a new ViewModel called the LoginViewModel.
2. The ones which define states like 'didLogin' should be renamed to 'isLoggedIn' for example.
3. 'canLogin' is not explicit in its purpose.
4. There should be a new purchaseState called 'login' on the BottomSheetViewModel.
5. The LoginViewModel should also control its state through a variable that has an Enum value, being that this Enum should have multiple values such as: 'choice', 'google', 'magic link'.

Reply:
Done